### PR TITLE
Fixes to mappings and refactor react, preact and solid into a factory with platform specific hooks

### LINKED ISSUE #928: [Bug]: Volar crash when using `new` keyword
### Issue Category

Other

### Bug Description

Found an issue with Volar and the `new` keyword.

#### Component scope `new`

Image

---

#### Module scope `new`

Image

### Reproduction Steps

N/A

### Environment

```markdown
- "TSRX for VSCode" v1.0.4
```

### Additional Context

_No response_

### Checklist

- [x] I've searched for existing issues
- [ ] I'm willing to help implement a fix

### PR DESCRIPTION
**Pull request by leonidaz**
If we're ok with this approach, we can see if we can refactor ripple transformers into the same structure and separate client and tsx transformers or maybe just the tsx.  The ripple server transformer is kind unique so not sure.  Anyway, I haven't looked into it yet.  But react, preact and solid definitely had a bunch in common.

Also, separated the mappings and shareable compile tests into shared tests that all 3 would run.  There may be others that could be shared but couldn't identify them all.  A great start that we can expand on.

Fixes and closes: #928, #923, #929, and #932

---

> [!NOTE]
> **Medium Risk**
> Medium risk: refactors the core JSX transform and source-mapping pipeline across React/Preact/Solid, which could subtly change emitted code and Volar mappings. Scoped-CSS and formatting fixes are covered by expanded shared test harnesses but still touch compiler/runtime-adjacent paths.
> 
> **Overview**
> **Unifies React/Preact/Solid compilation behind a shared `createJsxTransform` factory** (configured via a `JsxPlatform` descriptor) so JSX emission, import injection, scoped-class annotation, and source-map behavior are consistent across targets.
> 
> **Hardens Volar/source-mapping generation** by adding a shared source-mapping test harness and fixing mapping crashes/edge cases across many AST shapes (including TS predicates/generics, class methods, templates, and various statements/expressions).
> 
> **Fixes fragment + `<tsx>` behavior end-to-end:** the Prettier plugin now preserves `<>...</>` shorthand when printing `Tsx` nodes, and Ripple scoped CSS now applies inside `<tsx>...</tsx>` blocks and bare fragments (with new client/server tests). Also includes small Ripple typing/tsconfig cleanups (e.g., `RippleObject<T>` type tweak and proxy return casts).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 0d8ed65a33308ed0f3220186d3b9dca571f32a06. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/tsrx-solid/src/transform.js [hunk 1]
```diff
@@ -37,149 +45,71 @@ import {
...
+		// import injection goes through `hooks.injectImports`.
+		suspense: 'solid-js',
+		errorBoundary: 'solid-js',
+	},
+	jsx: {
+		rewriteClassAttr: false,
+		acceptedTsxKinds: ['solid'],
+	},
+	validation: {
+		requireUseServerForAwait: true,
```
[the remark this answers is not in the document]
Comment (leonidaz) [reply packages/tsrx-solid/src/transform.js:76]: @cursor we just changed it to true from false based on your previous feedback, what solution do you suggest?

**Review by trueadm — APPROVED** [review]
Awesome
